### PR TITLE
Clean up WORKSPACE.

### PR DESCRIPTION
**Pull request by mathetake**
Clean up WORKSPACE file:

- move all http_archives to `bazel/repositories.bzl`
- stop using `git_repository` for `protobuf` and `proxy_wasm_cpp_sdk`
- update cpp_sdk to the latest
- move ` test/test_data/wasm.bzl`  to `bazel/wasm.bzl`

Signed-off-by: mathetake <takeshi@tetrate.io>

### CODE REVIEW

**Review by lizan — APPROVED** [review]

File: bazel/dependencies_extra.bzl [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+load("@io_bazel_rules_rust//:workspace.bzl", "rust_workspace")
+load("@io_bazel_rules_rust//rust:repositories.bzl", "rust_repositories")
+load("@bazel_skylib//:workspace.bzl", "bazel_skylib_workspace")
+load("@proxy_wasm_cpp_host//bazel/cargo:crates.bzl", "proxy_wasm_cpp_host_raze__fetch_remote_crates")
+
+def dependencies_extra():
+    bazel_skylib_workspace()
+    rust_repositories()
+    rust_workspace()
```
Comment (PiotrSikora) [inline bazel/dependencies_extra.bzl:23]: I don't think we need `bazel_skylib_workspace` nor `rust_workspace` for `rules_rust` anymore, see: https://bazelbuild.github.io/rules_rust/#setup.

**Review by PiotrSikora — CHANGES_REQUESTED** [review]

File: bazel/dependencies_extra.bzl (hunk 1 above)
Comment (mathetake) [reply bazel/dependencies_extra.bzl]: done

**Review by PiotrSikora — APPROVED** [review]

**Comment by PiotrSikora** [comment]
Looks like `rules_rust` update resulted in a real issue:
```
warning: unused variable: `val`
  --> test/test_data/trap.rs:20:28
   |
20 | pub extern "C" fn trigger2(val: i32) -> i32 {
   |                            ^^^ help: if this is intentional, prefix it with an underscore: `_val`
   |
   = note: `#[warn(unused_variables)]` on by default
```
Could you fix that?

**Review by PiotrSikora — COMMENTED** [review]
Nit: I have slight preference for `repositories` and `dependencies` instead of `dependencies` and `dependencies_extra`.

**Review by PiotrSikora — APPROVED** [review]
Thanks!

**Pull request merged**